packages/whenable/src/whenable-utils.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+import { getTag } from '@endo/pass-style';
+
+/**
+ * @template T
+ * @param {any} specimen
+ * @returns {import('./types').WhenablePayload<T> | undefined}
+ */
+export const getWhenablePayload = specimen => {
```

Comment:
Are you purposely avoiding a `passStyleOf(specimen) === 'tagged'` check because it cannot pass such a check? What about checking at least `specimen[PASS_STYLE] === 'tagged'` ?

If it isn't really a tagged or a Passable, why pretend it is?

Reply:
I now understand that it is a proper passable tagged, so please add the `passStyleOf` check.

Reply:
I'm wary of `passStyleOf` because I thought it requires hardening.  I would expect that the whenable library shouldn't enforce that the promise fulfilments are hardened.  What is the best (though not necessarily foolproof) way to obtain the payload without requiring that `specimen` as hardened?

A snippet of actual code to define this `isWhenable` boolean would be much appreciated.

Reply:
> shouldn't enforce that the promise fulfilments are hardened

Why? If the purpose is durability, non-hardened fulfillments obviously cannot be passable or durable.

In any case, if you still want to be tolerant of non-hardened fulfillments, then I would indeed avoid `passStyleOf`. But I really need to understand the motivation. It seems at odds with the entire purpose of whenables.

Reply:
I want `when` to be benign even if the promise passed to it has a mutable fulfilment.  The reason is probably apparent with the latest changes, which introduce implicit whenable support into a local implementation of `E`.

Reply:
Ah.  I was somehow thinking that `passStyleOf` would *perform* hardening, which is not what I wanted.  I see that it throws when given an unhardened specimen, so I'd have to catch that.  That satisfies my question.

Reply:
Perhaps use the existing `isPassable` from @endo/base-zone that currently internally also does this try/catch. But says

```js
/**
 * TODO implement an isPassable that does not rely on try/catch, and
 * move it to @endo/pass-style.
 * This implementation is just a standin until then
 */
```

Reply:
I see it's fixed. Resolving.